[GH-2454] : Implement binary predicate `relate`  for Geopandas

gauravbyte opened the pull request:
Resolves: #2454
Depends on: #2230

## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)


## Is this PR related to a ticket?

- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #2454 

## What changes were proposed in this PR?
Implement binary predicate `relate`  for Geopandas with test cases

## How was this patch tested?
 Added tests

## Did this PR include necessary documentation updates?
- Yes, I have updated the documentation.

petern48 commented on python/tests/geopandas/test_geoseries.py:
```diff
@@ -1954,6 +1954,83 @@ def test_contains(self):
+        # 3. Test with align=False
+        result = s3.relate(s4, align=False)
+        expected = pd.Series(["0FFFFFFF2", "0FFFFFFF2"], index=range(1, 3))
+        assert_series_equal(result.to_pandas(), expected)
```
```suggestion
        self.check_pd_series_equal(result, expected)
```

petern48 commented on python/sedona/spark/geopandas/geoseries.py:
```diff
@@ -79,6 +79,7 @@
         "is_valid_reason",
         "length",
         "make_valid",
+        "relate",
```
Just an fyi, we don't actually keep this list updated (in fact, I just filed an [issue](https://github.com/apache/sedona/pull/2455) to remove this). It doesn't hurt to update it, so it's fine to leave this change here.

petern48 reviewed (commented):
Welcome! Thanks for working on this. It looks great. I mainly just had two minor suggestions to improve testing, but otherwise it looks good.

petern48 commented:
<img width="458" height="32" alt="image" src="https://github.com/user-attachments/assets/d7e6b93e-b866-415b-87fb-da99bfdccc82" />

optional: I recommend setting up the `pre-commit` hook to auto-run the linter after each commit. It's as simple as running `make checkinstall` in the home directory (or running `pip install pre-commit` and `pre-commit install`). Otherwise, you'll have to remember to run pre-commit manually before submitting a PR (which I personally find easy to forget).

gauravbyte commented on python/tests/geopandas/test_geoseries.py:
```diff
@@ -1954,6 +1954,83 @@ def test_contains(self):
+        point = Point(0, 0)
+        result = s.relate(point)
+        expected = pd.Series(["0FFFFFFF2", "0FFFFFFF2", "FF10F0FF2"])
+        assert_series_equal(result.to_pandas(), expected)
```
Resolved in the commit https://github.com/apache/sedona/pull/2455/commits/6f14d51392d141fe0dc18077508b3e5da022131e

gauravbyte commented:
Thanks for the review, @petern48! 
I’ve addressed your comments in [6f14d51](https://github.com/apache/sedona/commit/6f14d51392d141fe0dc18077508b3e5da022131e)
, rerun the tests, and set up the pre-commit hook as suggested.
Please review at your convenience.
